Add Microsoft.Mission spec to public repo

jchavaherrera opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [x] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [x] Other, please clarify:
    - New resource provider in azure-rest-api-specs. This RP exists already in [azure-rest-api-specs-pr/RPSaaSMaster](https://github.com/Azure/azure-rest-api-specs-pr/pull/22581)

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [X] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [X] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

jchavaherrera committed:
Moving specs from private to public

The files are the same as the one in https://github.com/Azure/azure-rest-api-specs-pr/pull/20074/files#diff-4764111bf455581c775da5c4711ccd11e820292fc9c84957532fc5d278b3ea2d

jchavaherrera committed:
Added missing files

jchavaherrera committed:
Merge branch 'main' into 2024-11-01-preview-Microsoft-Mission

jchavaherrera committed:
Add only version 2024-01-01 preview

jchavaherrera committed:
Merge branch 'main' into 2024-11-01-preview-Microsoft-Mission

jchavaherrera committed:
Remove old versions

jchavaherrera committed:
add back the old versions

jchavaherrera committed:
Removing old versions

jchavaherrera committed:
Added older version of the API

jchavaherrera committed:
Merge branch 'main' into 2024-11-01-preview-Microsoft-Mission

jchavaherrera committed:
Added older version of API to Readme

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Microsoft.Mission](https://apiview.dev/Assemblies/Review/035621869fed45c0affc90bd4bbf4eb4?revisionId=d95ab6d0ba524d00909065119f48a560)

jchavaherrera committed:
Update version.tsp

jchavaherrera committed:
remove 2022 versions

jchavaherrera commented:
We are having the error [MISSING_APIS_IN_DEFAULT_TAG], this is being caused by a deprecated version of the API, we would like to have that error suppressed

jchavaherrera commented:
All the files in the PR are already in the Azure:RPaaSMaster branch in the private repo, https://github.com/Azure/azure-rest-api-specs-pr/tree/RPSaaSMaster/specification/mission

The latest version of the API in the private repo is on this PR
https://github.com/Azure/azure-rest-api-specs-pr/pull/20074

gary-x-li commented:
Can you please address the followings?
1. Seems you are adding all versions to the public repo, not just `2024-11-01-preview`. Is this intended?
2. Please provide a summary of the differences between this PR and [the approved one in private repo](https://github.com/Azure/azure-rest-api-specs-pr/pull/20074), if any. This way, we just need to review the delta.
3. Swagger LintDiff has many errors. Please address them before sending the PR back for review.

mikeharder committed:
Merge branch 'main' into 2024-11-01-preview-Microsoft-Mission

jchavaherrera committed:
Delete versions prior 2024-06-01-preview

jchavaherrera committed:
remove status 200 of delete operations

jchavaherrera committed:
remove status 200 from openapi.json

jchavaherrera committed:
remove status 200 from example files

jchavaherrera committed:
remove status 200 from EnclaveEndpoints_Delete.json

jchavaherrera committed:
remove the status 200 when compiling

msyyc commented on specification/mission/Mission.Management/model.tsp:
```diff
@@ -0,0 +1,527 @@
+
+@doc("Virtual Enclave Interface")
+@armResourceOperations
+interface VirtualEnclave {
```
It is a little weird that all operations are defined in a tsp file named `model.tsp`. Personally I think it is better to put all operations in a tsp file with more specific tsp file (e.g. `routes.tsp`)

tgoodyear replied:
Thanks for the suggestion, addressed in [a8d0e084](https://github.com/Azure/azure-rest-api-specs/pull/32359/commits/a8d0e084bb1d0864fd5f933fe0bb9027802e3542)

jchavaherrera committed:
change ArmResourcePatchAsync to ArnCustomPatch

jchavaherrera committed:
compile file with ArmCustomPatch

jchavaherrera committed:
rollback to ArmResourcePatch

jchavaherrera committed:
added version 2024-12-01

jchavaherrera committed:
remove status 200 from ver 2024-12-01

jchavaherrera committed:
Merge branch 'main' into 2024-11-01-preview-Microsoft-Mission

jchavaherrera committed:
change ARMResourcePatch for ARMResourcePatch

jchavaherrera committed:
add @parameterVisibility to all ocurrences of ArmCustomPatchAsync

jchavaherrera committed:
minor fix on ArmCustomPatchAsync

jchavaherrera committed:
compile the model

jchavaherrera committed:
added location to Community Update

jchavaherrera commented:
/azp run

jchavaherrera committed:
Merge branch 'main' into 2024-11-01-preview-Microsoft-Mission

tgoodyear committed:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into 2024-11-01-preview-Microsoft-Mission

jchavaherrera committed:
location to community_update

jchavaherrera committed:
Update Community_Update.json

jchavaherrera commented:
/azp run

jchavaherrera commented:
/azp run [Swagger ModelValidation]

jchavaherrera commented:
/azp run Swagger ModelValidation

jchavaherrera committed:
Merge branch 'main' into 2024-11-01-preview-Microsoft-Mission

tgoodyear committed:
Merge branch '2024-11-01-preview-Microsoft-Mission' of https://github.com/Azure/azure-rest-api-specs into 2024-11-01-preview-Microsoft-Mission

jchavaherrera committed:
Merge branch 'main' into 2024-11-01-preview-Microsoft-Mission

tgoodyear committed:
Merge branch '2024-11-01-preview-Microsoft-Mission' of https://github.com/Azure/azure-rest-api-specs into 2024-11-01-preview-Microsoft-Mission

jchavaherrera committed:
added location to CommunityEndpoints

jchavaherrera committed:
Added location on all update example files

jchavaherrera committed:
added location to virtualEnclave_update

jchavaherrera committed:
made rulecollection optional

jchavaherrera committed:
change propertires to optional

jchavaherrera committed:
change properties to optional

jchavaherrera committed:
Added suppression

jchavaherrera committed:
add suppression

tgoodyear committed:
Merge branch '2024-11-01-preview-Microsoft-Mission' of https://github.com/Azure/azure-rest-api-specs into 2024-11-01-preview-Microsoft-Mission

tgoodyear committed:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into 2024-11-01-preview-Microsoft-Mission

jchavaherrera committed:
rollback suppressions

jchavaherrera committed:
made properties optional

jchavaherrera committed:
Merge branch 'main' into 2024-11-01-preview-Microsoft-Mission

jchavaherrera committed:
update openapi file

tgoodyear committed:
Merge branch '2024-11-01-preview-Microsoft-Mission' of https://github.com/Azure/azure-rest-api-specs into 2024-11-01-preview-Microsoft-Mission

tgoodyear committed:
Update for tsp update

tgoodyear committed:
Update formatting

tgoodyear committed:
Use latest version v6 of types.json

tgoodyear committed:
Use latest version v6 of types.json

tgoodyear committed:
Update patch model

tgoodyear committed:
Update patch model

tgoodyear committed:
Update patch model

tgoodyear committed:
Update patch model

tgoodyear committed:
Update patch model

tgoodyear committed:
Update patch model. Revert location addition in examples

tgoodyear committed:
Update patch model

tgoodyear committed:
Update patch model

tgoodyear committed:
Update patch model

tgoodyear committed:
Update patch model. Fix location in examples.

tgoodyear committed:
Update patch model. Fix location in examples.

tgoodyear committed:
Format update

tgoodyear committed:
Format update. Use Azure.Core.azureLocation for location def in mrg

tgoodyear committed:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into 2024-11-01-preview-Microsoft-Mission

tgoodyear committed:
Fix visibility for mrg location

tgoodyear committed:
Remove 2024_10 and 2024_11

tgoodyear committed:
Remove 2024_10 and 2024_11 examples

tgoodyear committed:
Remove 2024_10 and 2024_11 from readme.md

tgoodyear committed:
Remove 2024_10 and 2024_11 from resource-manager

tgoodyear committed:
Use Update<T> instead of separately defining the patchable props

tgoodyear committed:
Use Update<T> on patchResources

tgoodyear committed:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into 2024-11-01-preview-Microsoft-Mission

tgoodyear committed:
Revert optional/required params to match ve-common and hope the Update<T> model handles it correctly.

tgoodyear committed:
Re-Enable PATCH optionality.

tgoodyear committed:
Add types for ARM resources

tgoodyear committed:
Fix examples

tgoodyear committed:
Add suppressions for AvoidAnonymousTypes

tgoodyear committed:
Fix examples for type updates. Add uuid type to uuid fields.

tgoodyear committed:
Fix examples for type updates.

tgoodyear committed:
Remove uuid type

tgoodyear committed:
Remove uuid type

tgoodyear committed:
Prettier

tgoodyear committed:
Prettier

tgoodyear committed:
Update suppressions

tgoodyear committed:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into 2024-11-01-preview-Microsoft-Mission

tgoodyear committed:
Move routes to routes.tsp

rkmanda reviewed (approved):
:shipit:

tgoodyear committed:
Feedback from engineering -- remove Endpoint, InternalConnection, ExternalConnection models and interfaces. Update Community to include _ as part of name validation.

tgoodyear committed:
Merge branch 'main' into 2024-11-01-preview-Microsoft-Mission

tgoodyear committed:
Merge branch 'main' of https://github.com/Azure/azure-rest-api-specs into 2024-11-01-preview-Microsoft-Mission

tgoodyear committed:
Fix examples

tgoodyear committed:
Merge branch '2024-11-01-preview-Microsoft-Mission' of https://github.com/Azure/azure-rest-api-specs into 2024-11-01-preview-Microsoft-Mission

tgoodyear committed:
Remove examples for deprecated resources

tgoodyear committed:
Merge branch 'main' into 2024-11-01-preview-Microsoft-Mission

tgoodyear committed:
Remove examples for deprecated resources

tgoodyear committed:
Remove examples for deprecated resources

tgoodyear committed:
Merge branch 'main' into 2024-11-01-preview-Microsoft-Mission

jchavaherrera committed:
Merge branch 'main' into 2024-11-01-preview-Microsoft-Mission

tgoodyear committed:
Merge branch 'main' into 2024-11-01-preview-Microsoft-Mission

jchavaherrera committed:
Merge branch 'main' into 2024-11-01-preview-Microsoft-Mission

tgoodyear committed:
Merge branch 'main' into 2024-11-01-preview-Microsoft-Mission

jchavaherrera committed:
Added version 2025-05-01

jchavaherrera committed:
Merge branch 'main' into 2024-11-01-preview-Microsoft-Mission

jchavaherrera committed:
added experimental-extensible-enums: true

jchavaherrera committed:
Update TypeSpec config for package naming conventions

Renamed TypeScript package directory and name to use 'arm-placeholder' instead of 'armplaceholder'. Added C# emitter configuration with Azure flavor, output directory, and namespace settings to align with SDK generation standards.

jchavaherrera committed:
Update SDK config for Virtual Enclaves service

Replaced placeholder values in tspconfig.yaml with Virtual Enclaves-specific package directories, namespaces, and service names for Go, Python, Java, TypeScript, and C# emitters. This aligns the configuration with the Virtual Enclaves service for correct SDK generation.

jchavaherrera committed:
Update Go SDK service-dir path in tspconfig.yaml

Changed the @azure-tools/typespec-go service-dir from 'sdk/resourcemanager/virtualenclaves/armvirtualenclaves' to 'sdk/resourcemanager/virtualenclaves' to correct the SDK output directory configuration.

jchavaherrera committed:
Update service-dir default to sdk/virtualenclaves

Changed the 'service-dir' default value in tspconfig.yaml from 'sdk/placeholder' to 'sdk/virtualenclaves' to reflect the correct SDK output directory.

jchavaherrera committed:
Update service-dir path in tspconfig.yaml

Changed the service-dir default value from 'sdk/virtualenclaves' to 'sdk/resourcemanager/virtualenclaves' to ensure correct SDK emission path.

jchavaherrera committed:
Update service-dir path in tspconfig.yaml

Changed the default value of service-dir from 'sdk/resourcemanager/virtualenclaves' to 'sdk/virtualenclaves' to reflect the correct SDK directory structure.
